src/requestforattestation/RequestForAttestation.ts

```diff
@@ -52,15 +54,110 @@ function generateHashTree(contents: object): object {
   return result
 }
 
-function verifyClaimerSignature(rfa: RequestForAttestation): boolean {
-  return verify(rfa.rootHash, rfa.claimerSignature, rfa.claim.owner)
+function verifyClaimerSignature(reqForAtt: RequestForAttestation): boolean {
+  return verify(
+    reqForAtt.rootHash,
+    reqForAtt.claimerSignature,
+    reqForAtt.claim.owner
+  )
 }
 
 function getHashRoot(leaves: Uint8Array[]): Uint8Array {
   const result = u8aConcat(...leaves)
   return hash(result)
 }
 
+function compressNonceAndHash(nonceHash: any[]): any {
+  if (nonceHash.length === 0) {
+    return {
+      hash: nonceHash[0],
+    }
+  }
+  return {
+    nonce: nonceHash[0],
+    hash: nonceHash[1],
+  }
+}
+
+export function compressClaimHashTree(
+  reqForAtt: IRequestForAttestation
+): IRequestForAttestation['claimHashTree'] {
+  const result = {}
+  const claimTree = reqForAtt.claimHashTree
+
+  Object.keys(claimTree).forEach(entryKey => {
+    result[entryKey] = Object.values(claimTree[entryKey])
+  })
```

Comment:
is forEach preferred over a [for loop](https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html)?

I have not tested this but this should be equivalent?

```js
for (let entryKey in claimTree) {
  ...
}
```

Reply:
The current method seems to be more verbose and fits with our current theme of keeping things that way. 

For easy of use and reading.